R/types-chr.R

```diff
@@ -86,10 +86,20 @@ chr <- function(x, encoding = NULL) {
 #' str_encoding(latin1)
 #' as_bytes(latin1)
 #' }
-as_character <- function(x, encoding = NULL) {
+as_character <- function(x, encoding = NULL) UseMethod("as_character")
```

Comment:
We've decided not to make anything generic in rlang. We'll have a `switch_type()` or `switchpatch()` function for dispatch but it's still in the PRs under review.